Nicer error message if GTEC_TOKEN is not defined

Currently if you forget to `export GTEC_TOKEN=xxxxxxxx` you get a non - intuitive error message (below) about MD5 sums. We can fix this very easily by adding a check for the environment variable.

```
  CMake Warning (dev) at CMakeLists.txt:17 (file):
    Unexpected argument: MD5=f1993aea6d7c79e669f84dfd1acb2237
  This warning is for project developers.  Use -Wno-dev to suppress it.

  CMake Error at CMakeLists.txt:31 (message):
    Error occurred during download of gAPI.h: "HTTP response code said error"
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,7 +13,8 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
 # We use the EXPECTED_HASH option to check if the file is already present to avoid unnecessary repeat downloads.
 
 if(NOT DEFINED ENV{GTEC_TOKEN})
-  message(FATAL_ERROR "Environment variable GTEC_TOKEN is not defined. Run `export GTEC_TOKEN=...` and try again.")
+  message(FATAL_ERROR "Environment variable GTEC_TOKEN is not defined. Run `export GTEC_TOKEN=...` and try again."
+  " See README for more instructions.")
 endif()
 
 set(GTEC_API_LOCATION https://raw.githubusercontent.com/sense-base/PRIVATE_gUSBamp-Linux-Driver-C-API/refs/heads/main)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
 # as these are likely to go wrong if there is an error with GTEC_TOKEN.
 # We use the EXPECTED_HASH option to check if the file is already present to avoid unnecessary repeat downloads.
 
-if (NOT DEFINED ENV{GTEC_TOKEN})
+if(NOT DEFINED ENV{GTEC_TOKEN})
   message(FATAL_ERROR "Environment variable GTEC_TOKEN is not defined. Run `export GTEC_TOKEN=...` and try again.")
 endif()
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,6 +12,10 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
 # as these are likely to go wrong if there is an error with GTEC_TOKEN.
 # We use the EXPECTED_HASH option to check if the file is already present to avoid unnecessary repeat downloads.
 
+if (NOT DEFINED ENV{GTEC_TOKEN})
+  message(FATAL_ERROR "Environment variable GTEC_TOKEN is not defined. Run `export GTEC_TOKEN=...` and try again.")
+endif()
+
 set(GTEC_API_LOCATION https://raw.githubusercontent.com/sense-base/PRIVATE_gUSBamp-Linux-Driver-C-API/refs/heads/main)
 
 file(DOWNLOAD

